src/containers/App/sessionActions.js

```diff
@@ -7,6 +7,123 @@
  */
 
 import { createAction } from 'redux-actions';
+import { routerActions } from 'react-router-redux';
+import auth0 from 'auth0-js';
+import { applicationError } from '../../containers/Messages/messagesActions';
+import { fetchNewToken, isTokenValid } from '../../util/tokens';
+import { decodeIdToken, getTimeToUpdateInMs } from '../../util/jwtHelper';
+import { locationOrigin, auth0ClientId, auth0Domain, getToken } from '../../util/authHelpers';
 
+export const setAuthenticated = createAction('SET_AUTHENTICATED');
 export const setAccessToken = createAction('SET_ACCESS_TOKEN');
-export const fetchAccessToken = createAction('FETCH_ACCESS_TOKEN');
+export const setUserData = createAction('SET_USER_DATA');
+export const logoutAction = createAction('LOGOUT_ID_TOKEN');
+export const setIdToken = createAction('SET_ID_TOKEN');
+
+
+const auth = new auth0.WebAuth({
+  clientID: auth0ClientId || '',
+  domain: auth0Domain || '',
+  responseType: 'token id_token',
+  redirectUri: `${locationOrigin}/login/success`,
+  scope: 'openid app_metadata name',
+});
+
+export function parseHash(hash) {
+  return (dispatch) => {
+    auth.parseHash({ hash, _idTokenVerification: false }, (err, authResult) => {
+      if (authResult && authResult.accessToken && authResult.idToken) {
+        dispatch(setIdToken(authResult.idToken));
+        dispatch(setAuthenticated(true));
+        dispatch(setUserData(decodeIdToken(authResult.idToken)));
+        dispatch(routerActions.replace('/search'));
+      }
+    });
+  };
+}
+
+export function loginSocialMedia(type) {
+  auth.authorize({
+    connection: type,
+    client_id: auth0ClientId,
+  });
+}
+
+function doLogout(federated) {
+  return dispatch => fetchNewToken()
+    .then((token) => {
+      dispatch(setAccessToken(token.access_token));
+      dispatch(setAuthenticated(false));
+      dispatch(logoutAction());
+      auth.logout({
+        returnTo: `${locationOrigin}/`,
+        client_id: auth0ClientId,
+        federated,
+      });
+    })
+    .catch(err => dispatch(applicationError(err)));
+}
+
+export function logout() {
+  return doLogout(false);
+}
+
+export function logoutFederated() {
```

Comment:
Nå går det bare rundt og rundt i ring; ser hvorfor jeg jeg gikk for å løse dette med routes i stedet alt bare ble fungerende med en gang :)